server/services/lobby_service.go

```diff
@@ -84,6 +85,33 @@ func (l *LobbyServiceImpl) JoinLobby(msg pogo.LobbyJoinMsg) (Lobby, error) {
 	return *lobby, nil
 }
 
+// LeaveLobby will remove the player to the lobby, if it exists, or an error
+func (l *LobbyServiceImpl) LeaveLobby(lobbyID string, msg pogo.LobbyLeaveMsg) (Lobby, error) {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+
+	lobby, ok := l.activeLobbies[lobbyID]
+	if !ok {
+		return Lobby{}, fmt.Errorf("no lobby found with ID '%v'", lobbyID)
+	}
+
+	playerNum := -1
+	for i, p := range lobby.Players {
+		if p == msg.PlayerID {
+			playerNum = i
+			break
+		}
+	}
+
+	if playerNum == -1 {
+		return Lobby{}, fmt.Errorf("lobby '%v' does not have player '%v'", lobbyID, msg.PlayerID)
+	}
+
+	delete(lobby.ActiveConnections, msg.PlayerID)
```

Comment:
Does this actually kill the websocket connection?

Reply:
hmm... no, no it doesn't. I'll add an actual call to close the socket.